examples/pykokkos/streams.py

```diff
@@ -0,0 +1,27 @@
+import pykokkos as pk
+
+import cupy as cp
+
+@pk.workunit
+def print_stream(i, x):
+    if x == 0:
+        pk.printf("Default stream\n")
+    elif x == 1:
+        pk.printf("Stream 1\n")
+    elif x == 2:
+        pk.printf("Stream 2\n")
+
+if __name__ == "__main__":
+    space = pk.Cuda
+
+    s1 = cp.cuda.Stream()
```

Comment:
Should we use pk.Stream to avoid importing cupy

Reply:
The motivation here is that users might have pre-existing CuPy code they want to integrate with pykokkos, and this example shows how.

Reply:
but the example only uses cupy to create streams not to use cupy arrays; would be good to then use cupy for all parts